14/umbraco-cms/reference/custom-swagger-api.md

```diff
@@ -0,0 +1,201 @@
+---
+description: Example of a Custom API with Authorization and Swagger
+---
+
+# Custom API with Authorization and Swagger
+
+This article covers how to create a Custom API controller protected by the backoffice authorization policy. It also shows how to enable authorization UI in Swagger docs.
+
+
+{% hint style="info" %}
+
+Before  proceeding, make sure to read the [Management API](./management-api/README.md) article. It provides information about the Swagger documentation and Authorization used in this article.
+
+{% endhint %}
+
+This example can be a starting point for creating a secure custom API in the Swagger documentation or using the API Controller. You can find other examples in the [API versioning and OpenAPI](./api-versioning-and-openapi.md) article.
+
+1. Create a new `.cs` file called `MyBackOfficeSecurityRequirementsOperationFilter` at the root of the project in an Umbraco v14+ installation.
+
+2. Add the following code so that the new API shows in the Swagger documentation and Swagger UI:
+
+{% code title="MyBackOfficeSecurityRequirementsOperationFilter.cs" lineNumbers="true" %}
+
+```csharp
+
+using Microsoft.Extensions.Options;
+using Microsoft.OpenApi.Models;
+using Swashbuckle.AspNetCore.SwaggerGen;
+using Umbraco.Cms.Api.Management.OpenApi;
+using Umbraco.Cms.Core.Composing;
+
+namespace Umbraco.Cms.Web.UI.New.Custom;
+
+//Necessary code for the new API to show in the Swagger documentation and Swagger UI
+public class MyBackOfficeSecurityRequirementsOperationFilter : BackOfficeSecurityRequirementsOperationFilterBase
+{
+    protected override string ApiName => "my-api-v1";
+}
+
+public class MyConfigureSwaggerGenOptions : IConfigureOptions<SwaggerGenOptions>
+{
+    public void Configure(SwaggerGenOptions options)
+    {
+        options.SwaggerDoc("my-api-v1", new OpenApiInfo { Title = "My API v1", Version = "1.0" });
+        options.OperationFilter<MyBackOfficeSecurityRequirementsOperationFilter>();
+    }
+}
+
+public class MyComposer : IComposer
+{
+    public void Compose(IUmbracoBuilder builder)
+        => builder.Services.ConfigureOptions<MyConfigureSwaggerGenOptions>();
+}
+
+```
+
+{% endcode %}
+
+- The `BackOfficeSecurityRequirementsOperationFilter` enables backoffice authentication for Swagger UI.
+- `BackOfficeSecurityRequirementsOperationFilterBase` is an abstract base class that can be reused for custom APIs.
```

Comment:
```suggestion
- Our filter inherits from `BackOfficeSecurityRequirementsOperationFilterBase`. This marks our API as needing authorization and adds basic authentication and authorization metadata to all endpoints in our new API.
- `MyConfigureSwaggerGenOptions` configures our API swagger docs with our filter applied.
- `MyComposer ` makes sure the swagger generator knows about our API docs configuration at runtime.
```

Reply:
Careful - when reading this, one might think that the filter enables auth for the API endpoints... when in fact it just enabled auth in Swagger for the API.

Maybe something like this:

Our filter inherits from `BackOfficeSecurityRequirementsOperationFilterBase`. This marks our API as supporting authorization via Swagger.

Reply:
Applied, thank you for your suggestion 🙌